python/pyspark/ml/wrapper.py

```diff
@@ -87,9 +87,19 @@ def _new_java_array(pylist, java_class):
           - bool -> sc._gateway.jvm.java.lang.Boolean
         """
         sc = SparkContext._active_spark_context
-        java_array = sc._gateway.new_array(java_class, len(pylist))
-        for i in xrange(len(pylist)):
-            java_array[i] = pylist[i]
+        java_array = None
+        if len(pylist) > 0 and isinstance(pylist[0], list):
+            inner_array_length = 0
+            for i in xrange(len(pylist)):
+                inner_array_length = max(inner_array_length, len(pylist[i]))
```

Comment:
Does this mean if given pylist is [[1], [1], [1, 2, 3]], we will allocate array with length 3 for all inner arrays?

Reply:
Yes. I couldn't figure out a better way to do this. Any suggestions?

Reply:
I'm not sure now. Let me look at it tomorrow.

Reply:
Seems so. It can't be in different length.

For its usage to create `StringIndexerModel`, looks like it doesn't have bad effect. I'm not sure if it is ok for all cases. For example, if java side might get unexpected items (null?) from array. We need to add a note to this document.

Reply:
Java 2D arrays can be 'jagged', with different sizes in one dimension. It would be very unusual to create such a thing. I think it's fine to assume the input isn't jagged (or check it). Yes a little doc would be great.